prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -118,6 +122,78 @@ class CheckFunction : Logging {
         return HttpUtilities.httpResponse(request, responseBody.joinToString("\n") + "\n", httpStatus)
     }
 
+    /** @param code: "success" or "fail" enum for result of check's json response.
+     * want to limit to just one of the other */
+    enum class CheckResultsEnum(val code: String) { success("success"), fail("fail") }
+
+    /**
+     * Checks sftp settings works for a receiver.
+     *
+     * Differs from CheckFunction's `check()`:
+     *  - Uses OKTA oauth.
+     *  - ONLY supports POST
+     *  - ONLY does a single setting
+     *  - Does not support sending random data to a receiver
+     *  - Org and Setting names are split into two parameters
+     *  - Move CGI parameter into path
+     *  - Reply is 200 for check success and check failure cases.
+     *  - Reply is always in json since this is a REST api.
+     */
+    @FunctionName("checkreceiver")
+    fun checkreceiver(
+        @HttpTrigger(
+            name = "checkreceiver",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "checkreceiver/org/{orgName}/receiver/{receiverName}"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("orgName") orgName: String,
+        @BindingName("receiverName") receiverName: String
+    ): HttpResponseMessage {
+        val claims = AuthenticatedClaims.authenticate(request)
+
+        if (claims == null || !claims.authorized(setOf("*.*.primeadmin", "$orgName.*.admin"))) {
+            logger.warn("User '${claims?.userName}' FAILED authorized for endpoint ${request.uri}")
```

Comment:
It is beyond the scope of this ticket, but it feels like we should capture these kinds of auth failures somewhere else that we can review them later. They get lost in the log.

Reply:
Agreed. My hope is that we move to a decorator authenticator that includes a consistent logging approach so we can set up monitoring to catch them all.

Reply:
I created https://github.com/CDCgov/prime-reportstream/issues/7434 to track work on some kind of authenticator decorator feature.